frontend/model/contracts/group.js

```diff
@@ -262,30 +266,35 @@ DefineContract({
       }
     },
     'gi.contracts/group/proposal': {
-      validate: objectOf({
-        proposalType: proposalType,
-        proposalData: object, // data for Vue widgets
-        votingRule: ruleType,
-        expires_date_ms: number // calculate by grabbing proposal expiry from group properties and add to `meta.createdDate`
-      }),
-      process ({ data, meta, hash }, { state }) {
-        // make sure this isn't a duplicate proposal
+      validate: (data, { state, meta }) => {
+        objectOf({
+          proposalType: proposalType,
+          proposalData: object, // data for Vue widgets
+          votingRule: ruleType,
+          expires_date_ms: number // calculate by grabbing proposal expiry from group properties and add to `meta.createdDate`
+        })(data)
+
+        const dataToCompare = omit(data.proposalData, 'reason')
```

Comment:
This `dataToCompare` is only used once, in the `if`, and it would look cooler I think if there were two `omit` calls in the `if`, but if you think that makes the line too long I'm ok with keeping it as a variable (my preference though is to inline them both in the `if` though). However, if you decide you like it as a variable more, then can you move this line right above the `if` so that it's right next to the other call to `omit`? I feel like that would also help with readability/clarity.

Reply:
It’s used once but executed N times based on how many proposal there are. So instead of creating a new object every loop, we create it once and reuse it :)

Reply:
ah ok, great point!